components/builder-api/src/server/services/github.rs

```diff
@@ -408,3 +402,142 @@ async fn read_plan_targets(github: &GitHubClient,
 
     Ok(targets)
 }
+
+fn should_reject_build(plan: &PlanWithTarget, project: &Project, config: &Config, repo_url: &str) -> bool {
+    if repo_url != project.vcs_data {
+        warn!("Repo URL ({}) doesn't match project vcs data ({}). Aborting.",
+              repo_url, project.vcs_data);
+        return true;
+    }
+
+    if !config.api.build_targets.contains(&plan.1) {
+        debug!("Rejecting build with target: {:?}", plan.1);
+        return true;
+    }
+
+    if config.api.suppress_autobuild_origins.contains(&plan.0.origin.to_string()) {
+        debug!("Skipping autobuild for origin {:?}", &plan.0.origin);
+        return true;
+    }
+    return false
+}
+
+#[cfg(test)]
+mod test {
+
+    use crate::{db::models::projects::Project,
+                config::Config,
+                hab_core::{package::{target::{PackageTarget},
+                                     Plan}},
+                server::{services::github::{PlanWithTarget,
+                                            should_reject_build}}};
+    use std::str::FromStr;
+
+    #[test]
+    fn valid_plan_is_not_rejected() {
+        let mut config = Config::default();
+        config.api.suppress_autobuild_origins = vec!["hest".to_string(), "sest".to_string()];
+
+        let project = Project { id: 1,
```

Comment:
you might extract `create_project` and a `create_plan` functions that return a mutable project and a mutable plan and also have them take `origin`, `package`, `target`, and `repo_url` args. You could then reuse them in all the tests.

Reply:
But this will all live only in the mod test, I assume.

Reply:
Thanks for pointing me to the hab config apply testing.  I still would like to see that this succeeds inside of builder-api by observing whether or not the build kicks off depending on the current origin set defined in suppress_autobuild_origins.